diesel/src/connection/sqlite/mod.rs

```diff
@@ -0,0 +1,138 @@
+extern crate libsqlite3_sys as ffi;
+extern crate libc;
+
+#[doc(hidden)]
+pub mod raw;
+mod stmt;
+mod statement_iterator;
+mod sqlite_value;
+
+pub use self::sqlite_value::SqliteValue;
+
+use std::cell::Cell;
+use std::ffi::CStr;
+
+use backend::Sqlite;
+use query_builder::*;
+use query_builder::sqlite::SqliteQueryBuilder;
+use query_source::*;
+use result::*;
+use result::Error::QueryBuilderError;
+use self::raw::*;
+use self::stmt::*;
+use self::statement_iterator::StatementIterator;
+use super::{SimpleConnection, Connection};
+use types::HasSqlType;
+
+pub struct SqliteConnection {
+    raw_connection: RawConnection,
+    transaction_depth: Cell<i32>,
+}
+
+impl SimpleConnection for SqliteConnection {
+    fn batch_execute(&self, query: &str) -> QueryResult<()> {
+        self.raw_connection.exec(query)
+    }
+}
+
+impl Connection for SqliteConnection {
+    type Backend = Sqlite;
+
+    fn establish(database_url: &str) -> ConnectionResult<Self> {
+        RawConnection::establish(database_url).map(|conn| {
+            SqliteConnection {
+                raw_connection: conn,
+                transaction_depth: Cell::new(0),
+            }
+        })
+    }
+
+    fn execute(&self, query: &str) -> QueryResult<usize> {
+        try!(self.batch_execute(query));
+        Ok(self.raw_connection.rows_affected_by_last_query())
+    }
+
+    fn query_all<'a, T, U: 'a>(&self, source: T) -> QueryResult<Box<Iterator<Item=U> + 'a>> where
+        T: AsQuery,
+        T::Query: QueryFragment<Self::Backend>,
+        T::SqlType: 'a,
+        Self::Backend: HasSqlType<T::SqlType>,
+        U: Queryable<T::SqlType, Self::Backend>,
+    {
+        self.prepare_query(&source.as_query()).map(|stmt| {
+            Box::new(StatementIterator::new(stmt)) as Box<Iterator<Item=U>>
+        })
+    }
+
+    fn execute_returning_count<T>(&self, source: &T) -> QueryResult<usize> where
+        T: QueryFragment<Self::Backend>,
+    {
+        let stmt = try!(self.prepare_query(source));
+        try!(stmt.run());
+        Ok(self.raw_connection.rows_affected_by_last_query())
+    }
+
+    fn silence_notices<F: FnOnce() -> T, T>(&self, f: F) -> T {
+        f()
+    }
+
+    fn begin_transaction(&self) -> QueryResult<()>{
+        let transaction_depth = self.transaction_depth.get();
+        self.change_transaction_depth(1, if transaction_depth == 0 {
```

Comment:
I'd like to see the conditional here pulled out into a function, expression, local or something. It feels like using it as an inline expression here is trying to be "too clever". Thoughts?